setup.sh

```diff
@@ -0,0 +1,346 @@
+              echo -e "\033[32m::::::::::::: We want to install the tool tmux to do this. :::::::::::::\033[0m"
+              echo
+              echo -e "\033[32m::::::::::::: Run this command: sudo apt-get install tmux :::::::::::::\033[0m"
+          else
+              echo "OpenCRVS requires multiple terminal windows open in order to run OpenCRVS Core alongside the default country configuration."
+              echo
+              echo "We use the tool tmux to do this.  Please install it following the documentation here: https://github.com/tmux/tmux/wiki"
+          fi
+        fi
+        echo
+        echo -e "\033[32m::::::::::::::: After $i is installed please try again :::::::::::::::\033[0m"
+        echo
+        exit 1
+    fi
+done
+
+echo
+echo -e "\033[32m:::::: NOW WE NEED TO CHECK THAT YOUR NODE VERSION IS SUPPORTED ::::::\033[0m"
+echo
+#sleep 1
+myNodeVersion=`echo "$(node -v)" | sed 's/v//'`
+versionTest=$(do_version_check $myNodeVersion 14.15.0)
+if [ "$versionTest" == "LOWER" ] ; then
+  echo "Sorry your Node version is not supported.  You must upgrade Node to use a supported version."
+  echo "We recommend you install Node v.14.15.0, v14.15.4, v14.17.0 or v14.18.1 as this release has been tested on those versions."
+  echo "Documentation is here: https://nodejs.org/en/download/package-manager/#nvm"
+  echo "Then use nvm to install the Node version of choice.  For example run:\033[0m"
+  echo "nvm install 14.17.0"
+  exit 1
+  else
+    echo -e "Your Node version: $myNodeVersion is \033[32msupported!\033[0m :)"
+    echo
+fi
+
+if [ $OS == "UBUNTU" ]; then
+  echo
+  echo -e "\033[32m::::::::::::::::: Giving Docker user sudo privileges :::::::::::::::::\033[0m"
+  echo
+  echo -e "\033[32m::::::::::: Please enter your sudo password when prompted :::::::::::\033[0m"
+  echo
+  sudo chmod 666 /var/run/docker.sock
+  sudo usermod -aG docker $USER
+fi
+
+echo
+echo -e "\033[32m:::::::::::::::::::::: Initialising Docker Swarm ::::::::::::::::::::::\033[0m"
+echo
+echo -e "\033[32m:::::::: If Docker is already part of a swarm, this can fail. ::::::::\033[0m"
+echo
+echo -e "\033[32m::::::::::::::::: To leave an existing swarm, type: :::::::::::::::::\033[0m"
+echo
+echo "docker swarm leave --force"
+echo
+docker swarm init
+echo
+echo -e "\033[32m:::::::::: Stopping any currently running Docker containers ::::::::::\033[0m"
+echo
+if [[ $(docker ps -aq) ]] ; then docker stop $(docker ps -aq) ; fi
+
+echo
+echo -e "\033[32m:::::::::::::::::: Installing some Node dependencies ::::::::::::::::::\033[0m"
+echo
+npm install -g wait-on
+yarn install
+
+echo -e "\033[32m::::::::::::::::::::::: Creating some directories :::::::::::::::::::::::\033[0m"
+echo
+echo -e "\033[32m::::::::::::: Please enter your sudo password when prompted :::::::::::::\033[0m"
+echo
+openssl genrsa -out .secrets/private-key.pem 2048 && openssl rsa -pubout -in .secrets/private-key.pem -out .secrets/public-key.pem
+sudo mkdir -p data/elasticsearch
```

Comment:
I got this error on MacOS

```
Error response from daemon: error while creating mount source path '/host_mnt/Users/riku/Code/OpenCRVS/opencrvs-core/data/mongo': mkdir /host_mnt/Users/riku/Code/OpenCRVS/opencrvs-core/data/mongo: operation not permitted
error Command failed with exit code 1.
```

and I suspect it's because the data directory is owned by `root`. The next line changes the ownership to a user 1000, but at least my system doesn't have a user with that id. I checked this by running `dscl . -list /Users UniqueID`

The way I got this part working was I replaced both lines just with

```
mkdir -p data/elasticsearch
```

Reply:
Thanks Riku!